### PR TITLE
Docs: update content to work with website repo

### PR DESCRIPTION
**Pull request by robbymilo**
**What this PR does / why we need it**:
-Updates the docs to work with the website repo - mainly relative link updates.
-Adds a menu.yaml file to build the sidebar menu (no longer using front matter weight)

**Special notes for your reviewer**:
-Not tested with current docs site (only website repo)
-Docs will sync to the website repo via GitHub Action

### CODE REVIEW

File: docs/sources/administration/provisioning.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ In previous versions of Grafana, you could only use the API for provisioning dat
 
 ## Config File
 
-Checkout the [configuration](/installation/configuration) page for more information on what you can configure in `grafana.ini`
+Checkout the [configuration]({{< relref "../installation/configuration" >}}) page for more information on what you can configure in `grafana.ini`
```
Comment (marefr) [inline docs/sources/administration/provisioning.md]: This relative link looks weird - is the `../` needed? A lot more of those in the changes.

File: docs/sources/menu.yaml [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+    - link: /alerting/metrics/
+      name: Metrics
+    - link: /alerting/notifications/
+      name: Notifications
+  - name: Keyboard Shortcuts
+    link: /features/shortcuts/
+- name: What's New In Grafana
+  link: /whatsnew/
+  children:
+  - name: Version 6.4
```
Comment (marefr) [inline docs/sources/menu.yaml]: 6.5 is missing. You sure you've based your changes on the latest commit in master?

**Review by marefr — CHANGES_REQUESTED** [review]
Some comments. Not sure I like the idea of the yaml file. There's no chance of generating the yaml file based on frontmatter instead?. Frontmatter is also used for redirect/moved links etc so it will still be needed for other things besides menu.

File: docs/sources/menu.yaml (hunk 2 above)
Comment (robbymilo) [reply docs/sources/menu.yaml]: Yes, forgot to update the file.

File: docs/sources/administration/provisioning.md (hunk 1 above)
Comment (robbymilo) [reply docs/sources/administration/provisioning.md]: Unfortunately yes. In the website, these pages will be several directories deep, so referencing a page in another directory requires moving up a level.

**Comment by robbymilo** [comment]
> Some comments. Not sure I like the idea of the yaml file. There's no chance of generating the yaml file based on frontmatter instead?. 

The idea behind the yaml file for the menu structure is two fold:
- hugo does not work with multiple menus with the same name (we have duplicated pages using the same menu identifier across versions)
- we are trying to avoid scripts that modify content at build time

> Frontmatter is also used for redirect/moved links etc so it will still be needed for other things besides menu.

Aliases will still work in front matter.

**Review by daniellee — COMMENTED** [review]
@robbymilo can you give some tips on how to test this?

- I have tried testing with the website `old-docs` branch. Built the docs-base docker container there and then tested with `make watch` in the Grafana repo. This LGTM. All the paths and images seem to work as they should.

- I tested the master branch on the website repo instead of the `old-docs` branch and tested again with `make watch` - got a build error.

- I tested running hugo in the website repo after building the docs docker image on master but this didn't work either - got a blank page for http://localhost:3002/docs and the first page looked like this:
    image


- I tested building the docker image on the `old-docs` branch before running hugo in the website repo but same result as above.

**Comment by daniellee** [comment]
I should also mention that when testing the master (and old-docs) branch of the website repo with the `grafana/docs-base:latest` docker image (built with the old-docs branch) that I got loads of these errors:

```
ERROR 2019/11/28 15:42:01 error processing shortcode "_internal/shortcodes/relref.html" for page "docs/grafana/latest/features/datasources/cloudwatch.md": template: _internal/shortcodes/relref.html:1:76: exe
cuting "_internal/shortcodes/relref.html" at <relref .Page (.Get 0...>: error calling relref: No page found with path or logical name "../../reference/templating.md".
```

Am I doing something wrong?

**Comment by robbymilo** [comment]
@daniellee Can you try building the website on master? The `old-docs` branch will go away with this launch.

Can also see it here: https://staging.grafana.com/docs/

**Comment by torkelo** [comment]
Screen Shot 2019-11-29 at 07 41 01

Those links on the requirements page are still incorrect

**Comment by robbymilo** [comment]
> Screen Shot 2019-11-29 at 07 41 01
> 
> Those links on the requirements page are still incorrect

Those and a few others should now be fixed.

**Review by torkelo — APPROVED** [review]
Looks good now. After we have switched we could remove the menu sections from the markdown files.

**Review by marefr — CHANGES_REQUESTED** [review]
Can't get `cd docs && make watch` to work using website/master and `make docs-build`. How is local development of docs in Grafana repo suppose to work

**Review by marefr — APPROVED** [review]
LGTM

So new docs site is already live. Local dev support apparently coming soon

**Pull request merged**